pkg/deploy/template_chart_manager.go

```diff
@@ -0,0 +1,196 @@
+package deploy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"strings"
+	"sync"
+
+	"go.uber.org/zap"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1alpha1"
+	"github.com/kyma-project/lifecycle-manager/pkg/remote"
+	"github.com/slok/go-helm-template/helm"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8syaml "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultBufferSize = 2048
+	defaultFieldOwner = client.FieldOwner("lifecycle-manager")
+)
+
+type SKRWebhookTemplateChartManager struct {
+	config  *SkrChartManagerConfig
+	cache   *sync.Map
+	kcpAddr string
+}
+
+type SkrChartManagerConfig struct {
+	// WebhookChartPath represents the path of the webhook chart
+	// to be installed on SKR clusters upon reconciling kyma CRs.
+	WebhookChartPath           string
+	SkrWebhookMemoryLimits     string
+	SkrWebhookCPULimits        string
+	WatcherLocalTestingEnabled bool
+	GatewayHTTPPortMapping     int
+}
+
+func NewSKRWebhookTemplateChartManager(kcpRestConfig *rest.Config, config *SkrChartManagerConfig,
+) (*SKRWebhookTemplateChartManager, error) {
+	resolvedKcpAddr, err := resolveKcpAddr(kcpRestConfig, config)
+	if err != nil {
+		return nil, err
+	}
+	return &SKRWebhookTemplateChartManager{
+		config:  config,
+		cache:   &sync.Map{},
+		kcpAddr: resolvedKcpAddr,
+	}, nil
+}
+
+func (m *SKRWebhookTemplateChartManager) renderChartToRawManifest(ctx context.Context, kymaObjKey client.ObjectKey,
+	chartArgValues map[string]interface{},
+) (string, error) {
+	chartFS := os.DirFS(m.config.WebhookChartPath)
+	chart, err := helm.LoadChart(ctx, chartFS)
+	if err != nil {
+		return "", nil
+	}
+	return helm.Template(ctx, helm.TemplateConfig{
+		Chart:       chart,
+		ReleaseName: skrChartReleaseName(kymaObjKey),
+		Namespace:   metav1.NamespaceDefault,
+		Values:      chartArgValues,
+	})
+}
+
+func (m *SKRWebhookTemplateChartManager) cachedConfig(kymaObjKey client.ObjectKey,
+	chartArgValues map[string]interface{},
+) bool {
+	cachedArgValues, ok := m.cache.Load(kymaObjKey)
+	if !ok {
+		m.cache.Store(kymaObjKey, chartArgValues)
```

Comment:
why are you string chartConfig by kyma name? AFAIK, all kymas will have the same config. Shouldn't we simplify this just store config?